Modules/Core/Common/test/itkMatrixGTest.cxx

```diff
@@ -54,9 +54,35 @@ Expect_GetIdentity_returns_identity_matrix()
   EXPECT_TRUE(TMatrix::GetIdentity().GetVnlMatrix().is_identity());
 }
 
+
+template <typename TMatrix>
+constexpr bool
+vnl_matrix_is_convertible_to_itk_Matrix()
+{
+  return std::is_convertible<vnl_matrix<typename TMatrix::ValueType>, TMatrix>();
+}
+
+template <typename TMatrix>
+constexpr bool
+vnl_matrix_fixed_is_convertible_to_itk_Matrix()
+{
+  return std::is_convertible<
+    vnl_matrix_fixed<typename TMatrix::ValueType, TMatrix::RowDimensions, TMatrix::ColumnDimensions>,
+    TMatrix>();
+}
+
 } // namespace
 
 
+static_assert((!vnl_matrix_is_convertible_to_itk_Matrix<itk::Matrix<float>>()) &&
+                (!vnl_matrix_is_convertible_to_itk_Matrix<itk::Matrix<double, 4, 5>>()),
+              "itk::Matrix prevents implicit conversion from vnl_matrix");
```

Comment:
prevents -> doesn't prevent

Reply:
@dzenanz Well... it _does_ prevent implicit conversion from `vnl_matrix` 😸 But I think I understand what you mean, the static_assert message will appear when it doesn't 🤔  I'll make it "itk::Matrix _should_ prevent implicit conversion from vnl_matrix", hope that's OK to you!

And of course, I'll try to fix the CI failures!